src/CSnakes.Runtime/CPython/EventLoop.cs

```diff
@@ -52,6 +52,28 @@ private sealed class CancelRequest(CoroutineTask coroTask, CancellationToken can
         public CancellationToken CancellationToken { get; } = cancellationToken;
     }
 
+    private sealed class ScheduledTask(CoroutineTask task, CancellationTokenRegistration cancellationRegistration) : IDisposable
+    {
+        public CoroutineTask Task { get; } = task;
+
+        public void Cancel(CancellationToken cancellationToken) => Task.Cancel(cancellationToken);
```

Comment:
Why not dispose the registration here so it's done as early as possible?

```suggestion
        public void Cancel(CancellationToken cancellationToken)
        {
            cancellationRegistration.Dispose();
            Task.Cancel(cancellationToken);
        }
```